src/message_dao.py

```diff
@@ -12,27 +12,137 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from typing import Dict, Optional
+import json
+import os
+import subprocess
+import zlib
+
+from functools import lru_cache
+from typing import Dict, List, Optional
+
+from dotenv import load_dotenv
+from google.cloud.sql.connector import Connector
 from message import Message
 
+load_dotenv()
 EPOCH_HASH = 'ae9e7be4a03765456fe38287533e6446e8bbc93c'
 
 class MessageDao(object):
+    def __init__(self) -> None:
+        self._initialize_connection()
+        self._initialize_tables()
+
+    def _initialize_connection(self) -> None:
+        connector = Connector()
+        self.connection = connector.connect(
+            os.environ.get("HOST"),
+            "pymysql",
+            user = os.environ.get("USER"),
+            password = os.environ.get("PASSWORD")
+        )
+    
+    def _initialize_tables(self) -> None:
+        db_name = os.environ.get("DB")
+        if not db_name:
+            raise Exception("Missing environment variable for name of database.")
+        with self.connection.cursor() as cursor:
+            cursor.execute("CREATE DATABASE IF NOT EXISTS " + db_name)
+            self.connection.select_db(db_name)
+            # Mapping from message id to message
+            cursor.execute(
+                "CREATE TABLE IF NOT EXISTS Messages"
+                "(message_id VARCHAR(255) NOT NULL,"
+                "subject VARCHAR(255) NOT NULL,"
+                "from_ VARCHAR(255) NOT NULL,"
```

Comment:
nit: I don't remember why we use `from_` as the field name, but I don't think we need to use that naming in the DB?

Reply:
`from` is a keyword in python, which might be why it's named as such. I opted to maintain that naming scheme because `from` is also a keyword in SQL, so the two options are to avoid the keyword or to always use backticks when referring to the column. I opted for the former because it was already what we were doing within the python code.